src/index.d.ts

```diff
@@ -0,0 +1,395 @@
+         * @param  {(openedEvent:OpenedEvent)=>void} handler
+         * @returns void
+         */
+        setNotificationOpenedHandler(handler: (openedEvent: OpenedEvent) => void): void;
+
+        /**
+         * Prompts the iOS user for push notifications.
+         * @param  {(response:boolean)=>void} handler
+         * @returns void
+         */
+        promptForPushNotificationsWithUserResponse(handler: (response: boolean) => void): void;
+
+        /**
+         * Disable the push notification subscription to OneSignal.
+         * @param  {boolean} disable
+         * @returns void
+         */
+        disablePush(disable: boolean): void;
+
+        /**
+         * Disable or enable location collection (defaults to enabled if your app has location permission).
+         * @param  {boolean} shared
+         * @returns void
+         */
+        setLocationShared(shared: boolean): void;
+
+        /**
+         * Prompts the user for location permissions to allow geotagging from the OneSignal dashboard.
+         * @returns void
+         */
+        promptLocation(): void;
+
+        /**
+         * This method returns a "snapshot" of the device state for when it was called.
+         * @returns Promise<DeviceState>
+         */
+        getDeviceState(): Promise<DeviceState>;
+
+        /**
+         * Did the user provide privacy consent for GDPR purposes.
+         * @returns Promise<boolean>
+         */
+        userProvidedPrivacyConsent(): Promise<boolean>;
+
+        /**
+         * Tag a user based on an app event of your choosing so they can be targeted later via segments.
+         * @param  {string} key
+         * @param  {string} value
+         * @returns void
+         */
+        sendTag(key: string, value: string): void;
+
+        /**
+         * Tag a user wiht multiple tags based on an app event of your choosing so they can be targeted later via segments.
+         * @param  {object} tags
+         * @returns void
+         */
+        sendTags(tags: object): void;
+
+        /**
+         * Retrieve a list of tags that have been set on the user from the OneSignal server.
+         * @param  {(tags:object)=>void} handler
+         * @returns void
+         */
+        getTags(handler: (tags: object) => void): void;
+
+        /**
+         * Deletes a single tag that was previously set on a user.
+         * @param  {string} key
+         * @returns void
+         */
+        deleteTag(key: string): void;
+
+        /**
+         * Deletes multiple tags that were previously set on a user.
+         * @param  {string[]} keys
+         */
+        deleteTags(keys: string[]);
+
+        /**
+         * Allows you to set the user's email address with the OneSignal SDK.
+         * @param  {string} email
+         * @param  {string} authCode
+         * @param  {Function} handler
+         * @returns void
+         */
+        setEmail(email: string, authCode: string, handler: Function): void;
```

Comment:
`authCode` and `handler` should be optional

Reply:
Done.